roaringbitmap/src/main/java/org/roaringbitmap/ArrayBatchIterator.java

```diff
@@ -0,0 +1,28 @@
+package org.roaringbitmap;
+
+import static org.roaringbitmap.Util.toIntUnsigned;
+
+public class ArrayBatchIterator implements ContainerBatchIterator {
+
+  private int index = 0;
+  private final ArrayContainer array;
+
+  public ArrayBatchIterator(ArrayContainer array) {
+    this.array = array;
+  }
+
+  @Override
+  public int next(int key, int[] buffer) {
+    int consumed = 0;
+    short[] data = array.content;
+    while (consumed < buffer.length && index < array.getCardinality()) {
```

Comment:
I would be tempted to send ``array.getCardinality()`` and ``buffer.length`` to final local variables. Just to be safe performance-wise.

Reply:
I would be surprised if this made a difference. Since cardinality is not volatile, it is permissible to store the variable in a register. The method call will very likely be inlined. Nevertheless, I will try it, and I want to give this code the best chance against the alternative approach I have in mind.

Reply:
> I would be surprised if this made a difference.

So would I.